web/src/lottie-player.ts

```diff
@@ -0,0 +1,577 @@
+/*
+ * Copyright (c) 2023 the ThorVG project. All rights reserved.
+
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { html, PropertyValueMap, LitElement, type TemplateResult } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+
+// @ts-ignore: WASM Glue code doesn't have type & Only available on build progress
+import Module from '../dist/thorvg-wasm';
+import { THORVG_VERSION } from './version';
+
+let _tvg: any;
+let _module: any;
+(async () => {  
+  _module = await Module();
+  _tvg = new _module.TvgLottieAnimation();
+})();
```

Comment:
Loading WASM is too lazy, so if we sync with class initialize, rendering will hit slower.
I expect wasm loading gets done before being used, or we can provide option to choose lazy-load or pre-load later.

Reply:
> Loading WASM is too lazy, so if we sync with class initialize, rendering will hit slower.

Do you have any benchmarks? how much slower?

> I expect wasm loading gets done as faster as possible before being used, or we can provide option to choose lazy-load or pre-load later.

My understanding is, that everything will be preloaded once you render the web component, there is no difference. Only if you import globally I guess it will be preloaded early, but if a user would render or load the module programmatically? CMIIW. 

And, you can lazy-load or pre-load modules already using dynamic imports (if I understood you correctly)

Reply:
I'm worried this case:
> When user loads web-component after a long time since page loaded (like conditional rendering), then WASM loading will start at the time. It could be too lazy.

Yes, user can lazy-load with dynamic-import if they don't want to wasm initializing as page loaded.